bylaws/008-psr-evolution.md

```diff
@@ -0,0 +1,102 @@
+
+## Scope and objectives
+
+A PSR is often comprised of text and code, generally interfaces. Those interfaces are pieces of code that are released and tagged at a specific moment in time. However, the PHP language doesn't stand still; it evolves over time.
+
+This means that those interfaces need to both provide a stable contract, as well as evolve to leverage new language features that could help better enforce the behaviors described in the PSR itself.
+
+At the same time, a PSR cannot be changed after its release (at which point only erratas are allowed), to protect a package that declared compatibility from becoming de facto incompatible.
+
+This document defines a process to be followed in updating PSR interfaces, in a way that is not breaking in regard to behavior for end users, and with an appropriate upgrade path for the consumers.
+
+## Definitions
+
+ * **Consumer** - libraries and projects that consume one or more interfaces from the code released as part of the PSR in question.
+ * **Implementer** - libraries and projects that implement one or more interfaces from code released as part of the PSR in question.
+ * **Cross-compatibility** - the ability for a consumer or an implementer to support more than one code version of the PSR with a single release of their own.
+
+## New releases
+
+A new minor release of a PHP-FIG package containing interfaces for a PSR MUST follow these rules:
+ * the new release MUST follow [Semantic Versioning](https://semver.org/) rules;
+ * the PSR behavior MUST NOT be altered;
+ * consumers or implementers of those interfaces MUST NOT suffer breaking changes;
+ * the PHP version constraint of the PHP-FIG package MAY be increased to leverage newer language features, especially those that would aid cross-compatibility of consumers and implementers with the old and the new versions;
+ * the PHP version constraint of the PHP-FIG package MUST NOT be altered to use newer language features that would create cross-compatibility issues.
+ 
+A new major release of a PHP-FIG package containing interfaces for a PSR MUST follow the same rules, with this exception:
+ * the new major version of the package MAY contain breaking changes if the implementing packages have a reasonable upgrade path, like the possibility of releasing a cross-compatible implementation with the previous releases;
+ * the new major version of the package MAY refer to a new, superseding PSR.
+ 
+Note that if the upgrade path causes the consumers or implementers to maintain multiple versions of their libraries side-by-side, only to support multiple versions of the same PSR, the upgrade path is to be considered too steep.
+
+### Workflow
+
+Since releasing new versions of the interfaces MUST NOT alter the PSR in its behavior, those releases can be voted in with the same process as errata changes. The new releases MUST be declared and embedded with a brief explanation and a link in the PSR document, like in the following example:
+
+> \`\`\`php
```

Comment:
This does not render properly (at least on github) though.

Reply:
We lose the whitespace inside the code snippet, but the rest of the content is good:

![immagine](https://user-images.githubusercontent.com/6729988/68950491-8b44a800-07bc-11ea-9ef4-17a1d66e3e19.png)